### ISSUE TITLE
Barman 1.5.0 fails against PostgreSQL 8.4

### ISSUE THREAD

**Issue by gcalacoci** [2ndquadrant-it/barman#3]
Barman 1.5.0 does not work correctly with PostgreSQL 8.4, due to a check of the `wal_level` parameter.
The parameter does not exists on PostgreSQL 8.4.

 Manage PostgreSQL checks differently, always considering older versions.

**Comment by omacchioni** [comment]
+1 here, most likely linked to e2076b0273526a2a6ffc2eecc62e0aff3a3bf07b

The logs server-side show:

```
2015-10-20 23:09:16 CEST [2152.1] "" user=barman,db=postgres ERROR:  unrecognized configuration parameter "wal_level"
2015-10-20 23:09:16 CEST [2152.2] "" user=barman,db=postgres STATEMENT:  SHOW "wal_level"
2015-10-20 23:09:16 CEST [2152.3] "" user=barman,db=postgres ERROR:  current transaction is aborted, commands ignored until end of transaction block
2015-10-20 23:09:16 CEST [2152.4] "" user=barman,db=postgres STATEMENT:  SHOW "archive_mode"
[...]
```

A possible fix would be to do a "SHOW ALL;", store the result in a dict and fetch the values in the dict in get_pg_setting(), using sensible defaults when the run-time parameter doesn't exist.

**Comment by florentx** [comment]
I am afraid that it will still fail after 6d4bcd5, with PostgreSQL 9.4.

The fix should be implemented in the `get_remote_status` method, to skip fetching `SHOW "wal_level"`.

**Comment by gcalacoci** [comment]
We have fixed the issue with the 5d048c0 commit.
The fix is available in the just released 1.5.1b1 version.
